src/com/qamp/HarisJasarevic/homeworks/homework4/task2/Shape.java

```diff
@@ -0,0 +1,20 @@
+package com.qamp.HarisJasarevic.homeworks.homework4.task2;
+
+public abstract class Shape {
+
+    private final String color;
+
+    protected Shape(final String color) {
```

Comment:
Don't force `protected` too much. 

Think of it like this. If you were to create a new package, and wanted to use any of these Shapes, you would not been able to do it and that is not good.

Don't make methods protected to hide them, because of the design choice of protected to be visible in the same package.